scripts/crc-deploy.sh

```diff
@@ -0,0 +1,193 @@
+  echo "Pushing olm image"
+  docker tag "${LOCAL_OLM_IMAGE}" "${CRC_OLM_IMAGE}"
+  docker push "${CRC_OLM_IMAGE}"
+
+  # Tag and push registry image
+  echo "Pushing registry image"
+  docker tag "${LOCAL_OPM_IMAGE}" "${CRC_OPM_IMAGE}"
+  docker push "${CRC_OPM_IMAGE}"
+
+  # Create image streams
+  echo "Creating image streams: ${CRC_OLM_IMAGE} ${CRC_OPM_IMAGE}"
+  OLM_OPENSHIFT=image-registry.openshift-image-registry.svc:5000/openshift/olm:test
+  OPM_OPENSHIFT=image-registry.openshift-image-registry.svc:5000/openshift/opm:test
+  oc import-image olm --from="${OLM_OPENSHIFT}" --confirm
+  oc import-image opm --from="${OPM_OPENSHIFT}" --confirm
+}
+
+# make_manifest_patches takes in two parameters
+# OLM_IMG: the internal registry istag for the olm image
+# OPM_IMG: the internal registry istag for the opm image
+# it creates a helm values files and other manifest patch files applied by
+# scripts/generate_crds_manifests.sh
+function make_manifest_patches {
+  OLM_IMG=${1?Error:\ olm image undefined}
+  OPM_IMG=${2?Error:\ opm image undefined}
+  SED_OPTS=(-e 's#OLM_IMAGE#'"${OLM_IMG}"'#g' -e 's#OPM_IMAGE#'"${OPM_IMG}"'#g')
+
+  # helm values file
+  VALUES_TEMPLATE=$(mktemp)
+  cat << EOF > "${VALUES_TEMPLATE}"
+olm:
+  image:
+    ref: OLM_IMAGE
+catalog:
+  commandArgs: --configmapServerImage=OPM_IMAGE
+  opmImageArgs: --opmImage=OPM_IMAGE
+  image:
+    ref: OLM_IMAGE
+package:
+  image:
+    ref: OLM_IMAGE
+EOF
+
+  sed "${SED_OPTS[@]}" "${VALUES_TEMPLATE}" > "${CRC_E2E_VALUES}"
+
+  # psm operator patch
+  PSM_OPERATOR_PATCH_TEMPLATE=$(mktemp)
+  cat << EOF > "${PSM_OPERATOR_PATCH_TEMPLATE}"
+- command: update
+  path: spec.template.spec.containers[0].image
+  value: OLM_IMAGE
+- command: update
+  path: spec.template.spec.containers[0].env[1].value
+  value: OLM_IMAGE
+EOF
+
+  sed "${SED_OPTS[@]}" "${PSM_OPERATOR_PATCH_TEMPLATE}" > scripts/psm-operator-deployment.crc.e2e.patch.yaml
+
+  # collect-profiles patch
+  COLLECT_PROFILE_PATCH_TEMPLATE=$(mktemp)
+  cat << EOF > "${COLLECT_PROFILE_PATCH_TEMPLATE}"
+- command: update
+  path: spec.jobTemplate.spec.template.spec.containers[0].image
+  value: OLM_IMAGE
+EOF
+
+  sed "${SED_OPTS[@]}" "${COLLECT_PROFILE_PATCH_TEMPLATE}" > scripts/collect-profiles.crc.e2e.patch.yaml
+}
+
+# YQ for applying yaml patches
+YQ="go run ./vendor/github.com/mikefarah/yq/v3/"
+
+# CRC_E2E_VALUES is the name of the help values file
+# used to populate manifest templates with the locally built olm images
+export CRC_E2E_VALUES="values-crc-e2e.yaml"
+
+# Set kubeconfig to CRC if necessary
+export KUBECONFIG=${KUBECONFIG:-${HOME}/.crc/machines/crc/kubeconfig}
+KUBE_ADMIN_USER=$(crc console --credentials -o json | jq -r .clusterConfig.adminCredentials.username)
+KUBE_ADMIN_PASSWORD=$(crc console --credentials -o json | jq -r .clusterConfig.adminCredentials.password)
+
+# login to crc
+echo "Logging in as kubeadmin"
+oc login -u "${KUBE_ADMIN_USER}" -p "${KUBE_ADMIN_PASSWORD}" > /dev/null
```

Comment:
I just tried pulling down these changes, running locally, and I got hit with the following error:

```console
$ make crc
...
echo "Deploying OLM"
Deploying OLM
./scripts/crc-deploy.sh
Deploying OLM to CRC
Logging in as kubeadmin
error: couldn't get https://127.0.0.1:42973/.well-known/oauth-authorization-server: unexpected response status 403
make: *** [Makefile:180: crc-deploy] Error 1
```

Going to try and manually run the crc-deploy Makefile target and see whether that authentication issue is reproducible.

Reply:
Okay I figured out the issue: in the case where a user already has a defined $KUBECONFIG variable locally, and that kubeconfig contains many contexts', we're not changing the default context to the crc instance:

```console
$ echo $KUBECONFIG
/home/tflannag/.kube/config
$ export KUBECONFIG=${KUBECONFIG:-${HOME}/.crc/machines/crc/kubeconfig}
$ echo $KUBECONFIG
/home/tflannag/.kube/config
$
$ KUBE_ADMIN_USER=$(crc console --credentials -o json | jq -r .clusterConfig.adminCredentials.username)
$ KUBE_ADMIN_PASSWORD=$(crc console --credentials -o json | jq -r .clusterConfig.adminCredentials.password)
$ oc login -u "${KUBE_ADMIN_USER}" -p "${KUBE_ADMIN_PASSWORD}" > /dev/null
Logging in as kubeadmin
error: couldn't get https://127.0.0.1:42973/.well-known/oauth-authorization-server: unexpected response status 403
$ crc status
CRC VM:          Running
OpenShift:       Running (v4.10.3)
Podman:          
Disk Usage:      19.83GB of 32.74GB (Inside the CRC VM)
Cache Usage:     103.6GB
Cache Directory: /home/tflannag/.crc/cache
$ kubectx -c
kind-kind
$ # switch to the crc-admin context
$ kubectx crc-admin
$ oc login -u "${KUBE_ADMIN_USER}" -p "${KUBE_ADMIN_PASSWORD}" > /dev/null
$ echo $?
0
```